### PR TITLE
Topic branch for intergated documentation system

### PR DESCRIPTION
**Pull request by diekhans**
This commit seeds documentation topic branch

### CODE REVIEW

File: doc/source/intro_introduction.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+ReST protocols for data transfer are described `here <https://en.wikipedia.org/wiki/Representational_state_transfer>`_
+
+--------------------------
+The GA4GH web API
+--------------------------
+This API was created to enable researchers to better access and exchange genomic data across remote sites. Instead of downloading complete BAM files or
+whole genome annotations, the API allows retrieval of information on, for instance, single genes or genomic regions.
+
+For a full list of the GA4GA API goals, see :ref:`apigoals`
```
Comment (dglazer) [inline doc/source/intro_introduction.rst]: typo: GA4GH

**Comment by diekhans** [comment]
The current documentation can be viewed here:

    http://hgwdev.cse.ucsc.edu/~markd/ga4gh/documentation-pr/

File: doc/source/intro_introduction.rst (hunk 1 above)
Comment (diekhans) [inline doc/source/intro_introduction.rst]: @dglazer Pleaase feel free to edit away and commit to the topic branch.

Thanks!!!

**Comment by dglazer** [comment]
@diekhans , awesome to see this PR ready for review. I have a few process questions/suggestions before diving into the content.

* am I right that this PR has three types of content:
  * documentation infrastructure, that puts in place a system to generate and maintain an end-user-facing website
  * new overview documentation, contained in .rst files
  * suggested changes to schema-level comments in existing .avdl files

* if so, I suggest breaking it up into smaller pieces for easier review and quicker integration. Specifically:
  * **infrastructure**: if this were  pulled into a separate PR, that generated a skeleton website based on today's existing .avdl comments, I'd be +1 on it after a very quick sanity check
  * **overview**: I haven't looked at the new .rst files, but expect it will be relatively easy to get support for checking them in as a starting point, with a *PRELIMINARY -- NOT YET APPROVED* disclaimer. Then we can improve them individually with targeted PRs, with the goal of quickly getting each file to the point where people approve removing the disclaimer.
  * **schema**: similarly, I suggest breaking these into separate PRs (maybe by reads/variants/references?), so that people with strong opinions on narrow areas can review and resolve

Wdyt? I'm flexible on the details, but think we'll move a lot faster with something less monolithic.

**Comment by diekhans** [comment]
Hi @dglazer I agree your brake down is logical, however I think it would be more pragmatic for each of the task teams to update their parts of the documentation in the branch, both intro and schemes.  Essentially, this branch becomes where everything is staged until the whole thing is ready to merge to master.  

While I can see breaking in up into individual PRs, we are resource constrain at UCSC to do this.

Also, I think the real urgency here is to get the whole thing working, not to get the framework into master.  It would be better if there was solid outline for the other task teams to follow before they started.

It needs a lot of work before I would want it in master.

**Comment by dglazer** [comment]
Happy to try it that way; let's see how it goes. (I thought the PR-split would be less than an hour of work, but I don't have enough Git-fu to know.)

What process do you suggest for editing? Do you just want people to individually go in and make edits directly in the branch? If so, what policy do you suggest for conflicting opinions and edits?

**Comment by diekhans** [comment]
I suggest we actively discuss smaller disagreements in this ticket and larger issues get spun into their own ticket and maybe forked/PRed.  Keep it light-weight for small stuff.

I am still a fan of FIXMEs in the source too.  It makes it much easier to get an overall sense of a small issues in page than looking at individual tickets.  

Generally, the smaller issues, like unclear things in the avro, people can dive in.

The larger issue of how to structure the documentation, we need to really think on and maybe try out different organizations.  I would like to get fresh eyes,  not familiar with GA4GH, to look at it.

**Comment by jeromekelleher** [comment]
This looks fantastic, thanks @diekhans. Looking forward to seeing lots of PRs on this to make it ready for prime time.

**Pull request merged**